erpnext/controllers/accounts_controller.py

```diff
@@ -1224,15 +1225,38 @@ def update_child_qty_rate(parent_doctype, trans_items, parent_doctype_name, chil
 	def check_permissions(doc, perm_type='create'):
 		try:
 			doc.check_permission(perm_type)
-		except:
-			action = "add" if perm_type == 'create' else "update"
-			frappe.throw(_("You do not have permissions to {} items in a Sales Order.").format(action), title=_("Insufficient Permissions"))
+			validate_workflow_permission(doc)
+		except frappe.PermissionError:
+			actions = { 'create': 'add', 'write': 'update', 'cancel': 'remove' }
+
+			frappe.throw(_("You do not have permissions to {} items in a {}.")
+				.format(actions[perm_type], parent_doctype), title=_("Insufficient Permissions"))
+
+		except WorkflowPermissionError:
+			frappe.throw(_("You do not have workflow access to update this document."), title=_("Insufficient Workflow Permissions"))
+	
+	def validate_workflow_permission(doc):
+		workflow = get_workflow_name(doc.doctype)
+		if not workflow:
+			return
+
+		workflow_doc = frappe.get_doc("Workflow", workflow)
+		current_state = doc.get(workflow_doc.workflow_state_field)
+		roles = frappe.get_roles()
+
+		transitions = []
+		for transition in workflow_doc.transitions:
+			if transition.next_state == current_state and transition.allowed in roles:
+				if not is_transition_condition_satisfied(transition, doc):
+					continue
+				transitions.append(transition.as_dict())
```

Comment:
logic seems fine , but the transition condition is checked against an unsaved doc. 
So if user who can only approve upto **Rs.100**, uses Update Items and makes total **Rs.200**, `is_transition_condition_satisfied` is True as it considers grand total as 100 (previous version of doc as it is unsaved).
Think we need to check workflow after `save()`

Normally it wont give such a user the Approve/Reject Action button after user changes grand total to **Rs.200** and clicks on save. Hence user can't submit